spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java

```diff
@@ -44,29 +45,30 @@
 
 	private final RestTemplate restTemplate;
 
-	private final UriTemplate definitionsPath;
+	private final Link definitionsPath;
 
-	private final UriTemplate deploymentsPath;
+	private final Link deploymentsPath;
 
-	private final UriTemplate executionsPath;
+	private final Link executionsPath;
 
```

Comment:
This class should also grab the single-entity relations and use them later (see comments below)